packages/cli-hydrogen/src/cli/services/dev/check-version.ts

```diff
@@ -0,0 +1,40 @@
+import {system} from '@shopify/cli-kit'
+import {checkForNewVersion} from '@shopify/cli-kit/node/node-package-manager'
+import {renderInfo} from '@shopify/cli-kit/node/ui'
+
+const PACKAGE_NAME = '@shopify/hydrogen'
+
+export async function checkHydrogenVersion(directory: string): Promise<string | undefined> {
+  const currentVersion = await system.captureOutput(
+    'node',
+    ['-p', `require('./node_modules/${PACKAGE_NAME}/package.json').version`],
+    {
+      cwd: directory,
+    },
+  )
```

Comment:
Alternatively we could try to parse the lockfile itself but this seemed like a nicer alternative.